docs/rules/index.md

```diff
@@ -4,108 +4,106 @@ sidebarDepth: 0
-| [regexp/no-empty-character-class](./no-empty-character-class.md) | disallow character classes that match no characters |  |
-| [regexp/no-empty-group](./no-empty-group.md) | disallow empty group | :star: |
-| [regexp/no-empty-lookarounds-assertion](./no-empty-lookarounds-assertion.md) | disallow empty lookahead assertion or empty lookbehind assertion | :star: |
-| [regexp/no-escape-backspace](./no-escape-backspace.md) | disallow escape backspace (`[\b]`) | :star: |
-| [regexp/no-invalid-regexp](./no-invalid-regexp.md) | disallow invalid regular expression strings in `RegExp` constructors | :star: |
-| [regexp/no-lazy-ends](./no-lazy-ends.md) | disallow lazy quantifiers at the end of an expression | :star: |
-| [regexp/no-misleading-unicode-character](./no-misleading-unicode-character.md) | disallow multi-code-point characters in character classes and quantifiers | :wrench: |
-| [regexp/no-missing-g-flag](./no-missing-g-flag.md) | disallow missing `g` flag in patterns used in `String#matchAll` and `String#replaceAll` | :wrench: |
-| [regexp/no-optional-assertion](./no-optional-assertion.md) | disallow optional assertions | :star: |
-| [regexp/no-potentially-useless-backreference](./no-potentially-useless-backreference.md) | disallow backreferences that reference a group that might not be matched | :star: |
-| [regexp/no-super-linear-backtracking](./no-super-linear-backtracking.md) | disallow exponential and polynomial backtracking | :star::wrench: |
-| [regexp/no-super-linear-move](./no-super-linear-move.md) | disallow quantifiers that cause quadratic moves |  |
-| [regexp/no-useless-assertions](./no-useless-assertions.md) | disallow assertions that are known to always accept (or reject) | :star: |
-| [regexp/no-useless-backreference](./no-useless-backreference.md) | disallow useless backreferences in regular expressions | :star: |
-| [regexp/no-useless-dollar-replacements](./no-useless-dollar-replacements.md) | disallow useless `$` replacements in replacement string | :star: |
-| [regexp/strict](./strict.md) | disallow not strictly valid regular expressions | :star::wrench: |
+| Name                                                                                                                                       | Description                                                                                | 💼 | ⚠️ | 🔧 | 💡 | ❌  |
+| :----------------------------------------------------------------------------------------------------------------------------------------- | :----------------------------------------------------------------------------------------- | :- | :- | :- | :- | :- |
+| [confusing-quantifier](https://ota-meshi.github.io/eslint-plugin-regexp/rules/confusing-quantifier.html)                                   | disallow confusing quantifiers                                                             |    | ✅  |    |    |    |
```

Comment:
Is it possible to format the URL to be the original relative path + *.md?
This is because the SPA routing by vitepress will not work properly if we specify the URL.

```suggestion
| [confusing-quantifier](./no-contradiction-with-assertion.md)                                   | disallow confusing quantifiers                                                             |    | ✅  |    |    |    |
```

Reply:
The requirements for this are a bit confusing. Is the full rule URL only supposed to be used in README.md? I switched to using the full URL everywhere (using the `urlRuleDoc` option) so that users are sent to the external site regardless of what GitHub page they start from.

These are the places where we have rule links:
* README.md
* docs/rules/index.md
* docs/rules/{name}.md (some rules have links to their replacement links)

Reply:
> Is the full rule URL only supposed to be used in README.md?

Yes. Only README.md uses URLs. This is to link to the documentation page from the package page of npm. `docs/*` are files on the documentation page. They must be linked with relative paths.

If it's difficult, we can consider using a regexp to replace it after processing.

Reply:
I filed https://github.com/bmish/eslint-doc-generator/issues/295 to consider an option for that behavior. Do you think this is a common pattern for plugins that use a static site generator?

Reply:
I mostly use vuepress or vitepress so I'm not sure how other static site generators handle links😥

Reply:
I have addressed this by providing a `urlRuleDoc` function in `.eslint-doc-generatorrc.js` to ensure the URL is only used for rule doc links in `README.md` and otherwise the relative path is used. Please take a look.

https://github.com/bmish/eslint-doc-generator/releases/tag/v1.4.0

All feedback should be addressed at this point.

Reply:
Thank you, @bmish!

The relative paths are a bit suboptimal (they are basically `..`-ing out, just to go back into the same directory), but, aesthetics aside, this shouldn't be harmful.

Reply:
Yeah I have one bug filed https://github.com/bmish/eslint-doc-generator/issues/284 about resolving the paths better but luckily it still works.

Reply:
Nope, I take that back, they are harmful. The problem is that vitepress (which starts the local server) uses `docs` as the root directory, so we can't go out and back into it.

I'll make a new discussion for this.